Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -646,7 +646,7 @@ FontCascade::CodePath FontCascade::codePath(const TextRun& run, std::optional<un
     if (s_codePath != CodePath::Auto)
         return s_codePath;
 
-#if !USE(FREETYPE)
+#if !USE(FREETYPE) && !ENABLE(COMPLEX_TEXT_CONTROLLER_FOR_SIMPLE_CODE_PATH)
```

Comment:
That's interesting. This allows you to use the simple path for partial runs in places that guard against codePath not for shaping. I should test if something like this is possible in the general case as it could be a potential perf progression.

Reply:
As far as I know, the simple text code path doesn't support partial text run.
https://github.com/WebKit/WebKit/blob/eb8b277b0314108e5cb2886e39dfc52d98fc845a/Source/WebCore/platform/graphics/FontCascade.cpp#L1447-L1448